homeassistant/components/media_player/epson.py

```diff
@@ -0,0 +1,245 @@
+"""
+import asyncio
+import logging
+
+import aiohttp
+import voluptuous as vol
+from homeassistant.components.media_player import (
+    DOMAIN, MEDIA_PLAYER_SCHEMA, PLATFORM_SCHEMA, SUPPORT_NEXT_TRACK,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_SELECT_SOURCE, SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_HOST, CONF_NAME, CONF_PORT, CONF_SSL, STATE_OFF,
+    STATE_ON)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+
+REQUIREMENTS = ['epson-projector==0.1.1']
+
+DATA_EPSON = 'epson'
+DEFAULT_NAME = 'EPSON Projector'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=80): cv.port,
+    vol.Optional(CONF_SSL, default=False): cv.boolean
+})
+
+ATTR_CMODE = 'cmode'
+SUPPORT_CMODE = 33001
+
+SUPPORT_EPSON = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE |\
+            SUPPORT_CMODE | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP | \
+            SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK
+
+EPSON_SCHEMA = MEDIA_PLAYER_SCHEMA.extend({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Optional(ATTR_CMODE): cv.string
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+        hass,
+        config,
+        async_add_devices,
+        discovery_info=None):
+    """Set up the Epson media player platform."""
+    if DATA_EPSON not in hass.data:
+        hass.data[DATA_EPSON] = []
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+
+    try:
+        epson = EpsonProjector(
+            hass, name, host,
+            config.get(CONF_PORT), config.get(CONF_SSL))
+        await epson.update()
+    except (aiohttp.client_exceptions.ClientConnectorError,
+            asyncio.TimeoutError):
+        raise PlatformNotReady
+    if epson:
+        hass.data[DATA_EPSON].append(epson)
+        async_add_devices([epson], update_before_add=True)
+
+    async def async_service_handler(service):
+        """Handle for services."""
+        from epson_projector.const import (CMODE_LIST_SET)
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+        if entity_ids:
+            devices = [device for device in hass.data[DATA_EPSON]
+                       if device.entity_id in entity_ids]
+        else:
+            devices = hass.data[DATA_EPSON]
+        for device in devices:
+            if service.service == ATTR_CMODE:
+                cmode = service.data.get(ATTR_CMODE).lower()
+                if cmode in CMODE_LIST_SET:
+                    await device.select_cmode(cmode)
+            await device.update()
+    hass.services.async_register(
+        DOMAIN, ATTR_CMODE, async_service_handler,
```

Comment:
The service name should be prefixed with the platform name. The constant that holds the service name should start with `SERVICE_`.